src/test/scala/mesosphere/marathon/state/VolumeTest.scala

```diff
@@ -74,4 +75,26 @@ class VolumeTest extends MarathonSpec with Matchers {
       PersistentVolumeInfo(1024, `type` = DiskType.Mount, maxSize = Some(2048))).
       isSuccess shouldBe true
   }
+
+  test("validating that DiskSource asMesos converts to an Option Mesos Protobuffer") {
+    DiskSource(DiskType.Root, None).asMesos shouldBe None
+    val Some(pathDisk) = DiskSource(DiskType.Path, Some("/path/to/folder")).asMesos
+    pathDisk.getPath.getRoot shouldBe "/path/to/folder"
+    pathDisk.getType shouldBe Source.Type.PATH
+
+    val Some(mountDisk) = DiskSource(DiskType.Mount, Some("/path/to/mount")).asMesos
+    mountDisk.getMount.getRoot shouldBe "/path/to/mount"
+    mountDisk.getType shouldBe Source.Type.MOUNT
+
+    a[java.lang.IllegalArgumentException] shouldBe thrownBy {
```

Comment:
i don't think qualification matters here. the fact that there's a Some(path) at all is enough